src/structures/Interaction.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const Base = require('./Base');
+const { InteractionType } = require('../util/Constants');
+
+/**
+ * Represents an interaction, see {@link InteractionClient}.
+ * @extends {Base}
+ */
+class Interaction extends Base {
+  constructor(client, data) {
+    super(client);
+
+    /**
+     * Type of this interaction.
+     * @type {string}
+     * @readonly
+     */
+    this.type = InteractionType[data.type];
+
+    /**
+     * ID of this interaction.
+     * @type {Snowflake}
+     * @readonly
+     */
+    this.id = data.id;
+
+    /**
+     * The token of this interaction.
+     * @type {string}
+     * @readonly
+     */
+    this.token = data.token;
+
+    /**
+     * The application associated with this interaction.
+     * @type {Snowflake}
+     * @readonly
+     */
+    this.applicationID = data.application_id;
+
+    /**
+     * The channel this interaction was sent in.
+     * @type {?Channel}
+     * @readonly
+     */
+    this.channel = this.client.channels?.cache.get(data.channel_id) ?? null;
+
+    /**
+     * If this interaction was sent in a DM, the user which sent it.
+     * @type {?User}
+     * @readonly
+     */
+    this.user = data.user ? this.client.users?.add(data.user, false) : null;
```

Comment:
What if we don't have a users on the client? We can still return data via manually creating the entry 👁️